### PR TITLE
Add switching of tabs when executing add and budget commands

### LINKED ISSUE #270: Inconsistent tab switching behaviour
Currently, the `add-expense`/`add-income` commands do not switch to the respective tabs, while `add-bookmark-expense` and `add-bookmark-income` do. 

The `add-expense`/`add-income` commands should switch the app to the Expenses/Incomes tab respectively when executed. However, if executed on the Overview tab, the tab should not be switched.

The `set-expense-limit` and `set-savings-goal` commands should also switch the app to the Overview tab.

### PR DESCRIPTION
**Pull request by zhaojj2209**
Resolves #270.

Executing `add-expense` or `add-income` commands now switches the UI to the Expenses and Incomes tab respectively.

Executing `set-expense-limit` or `set-savings-goal` now switches the UI to the Overview tab.

This behaviour is now also documented in the user guide.

### CODE REVIEW

**Review by siddarth2824 — CHANGES_REQUESTED** [review]
There is inconsistent switching behaviour for the `set-expense-limit` and `set-savings-goal` command. These commands should ideally switch the tabs, to maintain consistency for tabs switching, to the `Overview` tab as the `Overview` tab consists information related to expense limit and savings goal.

**Review by yongping827 — CHANGES_REQUESTED** [review]
The removal of `assertCommandSuccess` from `SetExpenseLimitCommandTest` and `SetSavingsGoalCommandTest` seems unnecessary.

Consider implementing another overloaded `assertCommandSuccess` that has a parameter `Tab`. This way the `Tab` can be compared by `CommandResult::equals` rather than an additional `assertEquals`.

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/CommandTestUtil.java [hunk 1]
```diff
@@ -116,6 +117,16 @@ public static void assertCommandSuccess(Command command, Model actualModel, Comm
      * Convenience wrapper to {@link #assertCommandSuccess(Command, Model, CommandResult, Model)}
      * that takes a string {@code expectedMessage}.
      */
+    public static void assertCommandSuccess(Command command, Model actualModel, String expectedMessage,
+                                            Model expectedModel) {
```
Comment (yongping827) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/CommandTestUtil.java:121]:
```suggestion
            Model expectedModel) {
```
This should be the more appropriate indentation level, according to the standard. Ignore the indentation IntelliJ gives you.

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/CommandTestUtil.java [hunk 2]
```diff
@@ -116,6 +117,16 @@ public static void assertCommandSuccess(Command command, Model actualModel, Comm
...
+        CommandResult expectedCommandResult = new CommandResult(expectedMessage);
+        assertCommandSuccess(command, actualModel, expectedCommandResult, expectedModel);
+    }
+
+    /**
+     * Convenience wrapper to {@link #assertCommandSuccess(Command, Model, CommandResult, Model)}
+     * that takes a string {@code expectedMessage} and a boolean {@code isCalculateBudgetInfo}.
+     */
     public static void assertCommandSuccess(Command command, Model actualModel, String expectedMessage,
                                             Model expectedModel, boolean isCalculateBudgetInfo) {
```
Comment (yongping827) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/CommandTestUtil.java:131]:
```suggestion
            Model expectedModel, boolean isCalculateBudgetInfo) {
```

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/CommandTestUtil.java [hunk 3]
```diff
@@ -124,11 +135,12 @@ public static void assertCommandSuccess(Command command, Model actualModel, Stri
...
     /**
      * Convenience wrapper to {@link #assertCommandSuccess(Command, Model, CommandResult, Model)}
-     * that takes a string {@code expectedMessage}.
+     * that takes a string {@code expectedMessage}, a boolean {@code isCalculateBudgetInfo}
+     * and a Tab {@code tabToSwitchTo}.
      */
     public static void assertCommandSuccess(Command command, Model actualModel, String expectedMessage,
-            Model expectedModel) {
-        CommandResult expectedCommandResult = new CommandResult(expectedMessage);
+                                            Model expectedModel, boolean isCalculateBudgetInfo, Tab tabToSwitchTo) {
```
Comment (yongping827) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/CommandTestUtil.java:142]:
```suggestion
            Model expectedModel, boolean isCalculateBudgetInfo, Tab tabToSwitchTo) {
```

**Review by yongping827 — CHANGES_REQUESTED** [review]

**Review by ianyong — APPROVED** [review]
LGTM!

**Review by yongping827 — APPROVED** [review]
LGTM

**Review by siddarth2824 — APPROVED** [review]
LGTM!

**Pull request merged**